diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 2dd7df2..a47f176 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -26,7 +26,7 @@ if (NOT "${CPPA_NO_PROTOBUF_EXAMPLES}" STREQUAL "yes")
   find_package(Protobuf)
   if (PROTOBUF_FOUND)
     PROTOBUF_GENERATE_CPP(ProtoSources ProtoHeaders "${CMAKE_CURRENT_SOURCE_DIR}/remote_actors/pingpong.proto")
-    include_directories(${PROTOBUF_INCLUDE_DIR})
+    include_directories(${PROTOBUF_INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
     add_executable(protobuf_broker remote_actors/protobuf_broker.cpp ${ProtoSources})
     target_link_libraries(protobuf_broker ${CMAKE_DL_LIBS} ${CPPA_LIBRARY} ${PTHREAD_LIBRARIES} ${PROTOBUF_LIBRARIES})
     add_dependencies(protobuf_broker all_examples)
